src/Sulu/Bundle/AdminBundle/Resources/js/components/AutoComplete/AutoComplete.js

```diff
@@ -0,0 +1,182 @@
+// @flow
+import React from 'react';
+import type {ChildrenArray, Element, ElementRef} from 'react';
+import {observer} from 'mobx-react';
+import {action, computed, observable} from 'mobx';
+import debounce from 'debounce';
+import Input from '../Input';
+import Popover from '../Popover';
+import Menu from '../Menu';
+import Suggestion from './Suggestion';
+import autoCompleteStyles from './autoComplete.scss';
+
+const LENS_ICON = 'search';
+const DEBOUNCE_TIME = 250;
+const POPOVER_HORIZONTAL_OFFSET = 5;
+const POPOVER_VERTICAL_OFFSET = -2;
+
+type Props = {
+    children: ChildrenArray<Element<typeof Suggestion>>,
+    /** The value inside the input field */
+    value: string,
+    placeholder?: string,
+    /** Shows the loading indicator when true */
+    loading?: boolean,
+    /** Called on every input change */
+    onChange: (value: string) => void,
+    /** Debounced version of `onChange`. Put network requests inside of this handler */
+    onDebouncedChange: (value: string) => void,
+    onSuggestionSelection: (value: string) => void,
```

Comment:
I don't know if I would have an `onChange` and an `onSuggestionSelection` event... The thing is that the value should probably only change if it is a valid value from the available selections...